frontend/src/pages/enabledApplications/EnabledApplications.tsx

```diff
@@ -66,8 +66,10 @@ const EnabledApplications: React.FC = () => {
      */
     if (loaded && components.length) {
       _.difference(
-        components.map((c) => c.metadata.name),
-        enabledComponents.map((c) => c.metadata.name),
+        components.filter((component) => component.spec.isEnabled).map((c) => c.metadata.name),
+        enabledComponents
+          .filter((component) => component.spec.isEnabled)
+          .map((c) => c.metadata.name),
```

Comment:
Why do we need the filtering here? We should only get back enabled applications from call to `useWatchComponents(true)`

Reply:
I think what we get back is `shownOnEnabledPage` components, it could be either enabled or not, so we need to check the `isEnabled` flag and only fire events of those actually enabled applications.

Reply:
Above:
[const { components, loaded, loadError } = useWatchComponents(true);](https://github.com/red-hat-data-services/odh-dashboard/blob/46e04871ab731a4f77e2ac9a37e2d89c546a9397/frontend/src/pages/enabledApplications/EnabledApplications.tsx#L54)

That will only return enabled applications.

Reply:
If I understand correctly, `useWatchComponents(true)` will finally get the components returned by the `listComponents` function at the backend, right? There I changed the API here https://github.com/red-hat-data-services/odh-dashboard/pull/184/files#diff-a40e274495aa442e8b068789cdfe234d0c79eccffa8e3de2da8703d2b9b559d4R19
so it will return all `shownOnEnabledPage` components. Maybe I should do a filter when I set `enabledComponents = components` so we don't have to filter `enabledComponents` but we still need to filter `components` we got from `useWatchComponents(true)` hook.

Reply:
the `true` flag should tell the backend to only return what is shown on the `Enabled` page, if that is not the case then the backend needs to be fixed.

Reply:
Ah.... I get it now. This is only to send the tracking event when apps _become_ enabled. My confusion.

Reply:
That's right. 😉